iothub/service/src/Message.cs

```diff
@@ -101,127 +101,87 @@ internal Message(Stream stream, bool ownStream)
         /// + {'-', ':', '/', '\', '.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}.
         /// Non-alphanumeric characters are from URN RFC.
         /// </summary>
+        /// <remarks>
+        /// If this value is not supplied by the user, the service client will set this to a new GUID.
+        /// </remarks>
         public string MessageId
         {
-            get
-            {
-                return this.GetSystemProperty<string>(MessageSystemPropertyNames.MessageId);
-            }
-
-            set
-            {
-                this.SystemProperties[MessageSystemPropertyNames.MessageId] = value;
-            }
+            get => GetSystemProperty<string>(MessageSystemPropertyNames.MessageId) ?? Guid.NewGuid().ToString();
+            set => SystemProperties[MessageSystemPropertyNames.MessageId] = value;
         }
 
         /// <summary>
         /// [Required] Destination of the message
         /// </summary>
         public string To
         {
-            get
-            {
-                return this.GetSystemProperty<string>(MessageSystemPropertyNames.To);
-            }
-
-            set
-            {
-                this.SystemProperties[MessageSystemPropertyNames.To] = value;
-            }
+            get => GetSystemProperty<string>(MessageSystemPropertyNames.To);
+            set => SystemProperties[MessageSystemPropertyNames.To] = value;
         }
 
         /// <summary>
         /// [Optional] The time when this message is considered expired
         /// </summary>
         public DateTime ExpiryTimeUtc
         {
-            get
-            {
-                return this.GetSystemProperty<DateTime>(MessageSystemPropertyNames.ExpiryTimeUtc);
-            }
-
-            set
-            {
-                this.SystemProperties[MessageSystemPropertyNames.ExpiryTimeUtc] = value;
-            }
+            get => GetSystemProperty<DateTime>(MessageSystemPropertyNames.ExpiryTimeUtc);
+            set => SystemProperties[MessageSystemPropertyNames.ExpiryTimeUtc] = value;
         }
 
         /// <summary>
-        /// Used in message responses and feedback
+        /// A string property in a response message that typically contains the MessageId of the request, in request-reply patterns.
         /// </summary>
+        /// <remarks>
+        /// If this value is not supplied by the user, the service client will set this to <see cref="MessageId"/>.
```

Comment:
This is based on the ms doc description for c2d message correlation Id: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct#system-properties-of-c2d-iot-hub-messages